code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -9,14 +9,37 @@ function UserRoles() {
 // In line with other repository functions, this returns a promise,
 // but the model document is first converted to an object so the spread operator works as expected.
 function getRoles(userId) {
-  return UserRoles().findOne({ userId }).exec().then(document => document.toObject());
+  return UserRoles().findOne({ userId }).exec()
+    .then((document) => {
+      try {
+        return document.toObject();
+      } catch (err) {
+        return addEmptyRecordForNewUser(userId);
+      }
+    });
 }
 
 function getProjectUsers(projectKey) {
   const query = { 'projectRoles.projectKey': { $eq: projectKey } };
   return UserRoles().find(query).exec();
 }
 
+async function addEmptyRecordForNewUser(userId) {
+  // setOnInsert used to prevent accidental override of entry
```

Comment:
I'm in two minds about whether this should be an upsert or a create as the only way this gets executed is if we have already checked if a record exists and found it hasn't. For a create if could just be

```
const user = ...
return UserRoles().create(user);
```